dbcon/mysql/ha_calpont_execplan.cpp

```diff
@@ -5150,6 +5190,16 @@ void gp_walk(const Item* item, void* arg)
 
                 case STRING_RESULT:
                 {
+                    // Special handling for 0xHHHH literals
+                    const Type_handler *tph = item->type_handler();
+                    if (dynamic_cast<const Type_handler_hex_hybrid*>(tph) != NULL)
+                    {
+                        Item_hex_hybrid *hip = (Item_hex_hybrid*) item;
```

Comment:
I know that MDB code uses c-type cast but we should stick with cpp-style reinterpret_cast. Feel free to tell me if I use c-style cast accidently.

Reply:
Done. I could not directly use reinterpret_cast as the compiler complained about the constness of item (as it is passed as const Item* to gp_walk()). So I had to first cast away the constness using const_cast.